packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -436,6 +436,44 @@ const resolveObjectPatternValueNode = (
   return objectPatternValueNode;
 };
 
+/**
+ * Will return property key if it is different from its value and its value is
+ * equal to reference name for which we have to resolve binding.
+ *
+ * Eg. If we have something like `const { key: value } = { key: 'something' }`, and
+ * reference name is `value`, it will return `key` so that it can be resolved to
+ * 'something' otherwise it won't get resolved.
+ * Input: `node: const { key: value } = { key: 'something' }, referenceName: 'value'`
+ * Output: `'key'`
+ *
+ * Input: `node: const { key } = { key: 'something' }, referenceName: 'value'`
+ * Output: `'value'`
```

Comment:
why does this return value? should this not return `key`

Reply:
ah because we are saving it in result `let result = referenceName;` and if `keyName !== keyValue && keyValue === referenceName` condition fails, it will return the result with same value.